test/integration/dbversion.test.js

```diff
@@ -0,0 +1,54 @@
+describe('dbVersion integration test', () => {
+  const dbVersion = require('../../server/dbversion')
+  const dbVersionGetPending = dbVersion.getPending
+
+  async function truncateVersionTable () {
+    return dbVersion.umzug.storage.model.destroy({
+      where: {},
+      truncate: true
+    })
+  }
+
+  beforeEach(async () => {
+    await truncateVersionTable()
+  })
+
+  afterEach(() => {
+    dbVersion.getPending = dbVersionGetPending
+  })
+
+  test('dbVersion gets a list of available migrations', () => {
+    expect(dbVersion.availableVersions.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion validates version in database', async () => {
+    await dbVersion.umzug.storage.model.upsert({ name: dbVersion.highestVersion })
+    await dbVersion.refreshCurrentDatabaseVersion()
+    expect(dbVersion.currentDatabaseVersion).toBe(dbVersion.highestVersion)
+  })
+
+  test('dbVersion reports a list of pending migrations', async () => {
+    const pending = await dbVersion.getPending()
+    expect(pending.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion reports an issue if there are pending migrations', async () => {
+    await expect(dbVersion.throwAnyErrors()).rejects.toThrow('Pending Database migrations exist')
+  })
+
+  test('dbVersion reports an issue if there are no database versions stored', async () => {
+    dbVersion.getPending = jest.fn(() => { return [] })
+    await expect(dbVersion.throwAnyErrors()).rejects.toThrow('No database version could be found')
+  })
+
+  test('dbVersion does not report an issue if there is a valid database version', async () => {
+    dbVersion.getPending = jest.fn(() => { return [] })
+    await dbVersion.umzug.storage.model.upsert({ name: dbVersion.highestVersion })
+    await expect(dbVersion.throwAnyErrors()).resolves.not.toThrow()
+  })
+
+  test('dbVersion reports an issue if there are is an unknown database version', async () => {
+    await dbVersion.umzug.storage.model.upsert({ name: 'zzzzzzzzzzesttest.nofile' })
+    await expect(dbVersion.throwAnyErrors()).rejects.toThrow()
```

Comment:
Missing expected error message. This test is actually picking up the "Pending Database migrations exist" error because `getPending` isn't being mocked.

There is a complication in testing the unknown database version error because the message is dynamic, with part depending on the most recently created migration name. I would recommend using a regular expression to test only the first half of the message:
```
test('dbVersion reports an issue if there are is an unknown database version', async () => {
    dbVersion.getPending = jest.fn(() => { return [] })
    await dbVersion.umzug.storage.model.upsert({ name: 'zzzzzzzzzzesttest.nofile' })
    await expect(dbVersion.throwAnyErrors()).rejects.toThrow(/^Current database version \(zzzzzzzzzzesttest.nofile\) unknown to this code/)
  })
```

Reply:
Fixed now by using your code. Thanks for that.